File: Animator/C#/PatternsAnimator/Client/Square.cs
using PatternsAnimator.Lib;
using PatternsAnimator.Utils;

namespace PatternsAnimator.Client
{
    public class Square : IMovable, IRotateable
    {
        private Point _position;
        private double _angle;

        public Square()
        {
            _position = new Point(0, 0);
            _angle = 0;
        }

        public void SetPosition(Point p)
        {
            _position = p;
        }

        public void SetAngle(double a)
        {
            _angle = a;
        }

        public override string ToString()
        {
            return $"Square: x = {_position.X}, y = {_position.Y}, angle = {_angle}";
        }
    }
}

File: Animator/Java/src/Client/Square.java
package Client;

import Lib.IMovable;
import Lib.IRotatable;
import Utils.Point;

public class Square implements IMovable, IRotatable {
    private Point p;
    private double angle;

    public Square() {
        this.p     = new Point(0, 0);
        this.angle = 0.0;
    }

    @Override
    public void setPosition(Point p) {
        this.p = p;
    }

    @Override
    public void setAngle(double a) {
        this.angle = a;
    }

    @Override
    public String toString() {
        return "Square{" +
                "p=" + p +
                ", angle=" + angle +
                '}';
    }
}
